lib/utils/widgets/lerpy_hero/lerpy_hero.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:flutter/material.dart';
+import 'package:thc/utils/widgets/lerpy_hero/data_widget.dart';
+
+/// The [T] type can be as simple as a [Color] (using [Color.lerp]),
+/// or you can go wild with [Record]s, extension types, and custom classes.
+///
+/// {@template LerpyHero.extends_Widget}
+/// We're extending [Widget] instead of [StatelessWidget] so that we can have
+/// a private [_build] method.
+/// {@endtemplate}
+abstract class LerpyHero<T> extends Widget {
+  const LerpyHero({
+    required this.data,
+    this.transitionOnUserGestures = true,
+    this.tag,
+    this.child,
+    super.key,
+  });
```

Comment:
This class has the button animation logic, and we can reuse it for other things in the future too!